custom/git-app-deployer/main.go

```diff
@@ -62,31 +76,45 @@ func getKubeConfig() (*rest.Config, error) {
 
 //GetData derive the sock-shop filePath and timeout
 //it derive the filePath based on sock-shop scenario(week vs resilient)
-func GetData() (string, string, int) {
-	namespace := flag.String("namespace", "", "namespace for the application")
-	typeName := flag.String("typeName", "", "type of the application")
-	timeout := flag.Int("timeout", 300, "timeout for application status")
+func GetData() (vars *APPVars) {
+	appVars := APPVars{
+		namespace: "",
+		filePath:  "",
+		timeout:   0,
+		operation: "",
+	}
+
+	appVars.namespace = *flag.String("namespace", "", "namespace for the application")
+	appVars.filePath = *flag.String("typeName", "weak", "type of the application")
+	appVars.timeout = *flag.Int("timeout", 300, "timeout for application status")
+	appVars.operation = *flag.String("operation", "apply", "revert application type")
 	flag.Parse()
 
-	if *namespace == "loadtest" {
-		return *namespace, "load-test.yaml", *timeout
+	if appVars.namespace == "loadtest" {
+		vars.filePath = "load-test.yaml"
 	}
-	if *typeName == "" || *typeName == "weak" {
-		return *namespace, "weak-sock-shop.yaml", *timeout
+	if appVars.namespace == "sock-shop" {
+		vars.filePath = appVars.filePath + "-sock-shop.yaml"
 	}
-	return *namespace, *typeName + "-sock-shop.yaml", *timeout
+	return &appVars
 }
 
-// CreateNamespace creates a sock-shop namespace
+// CreateNamespace creates a sock-shop/loadtest namespace
 func CreateNamespace(clientset *kubernetes.Clientset, namespaceName string) error {
 	nsSpec := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}
 	_, err := clientset.CoreV1().Namespaces().Create(nsSpec)
 	return err
 }
 
-//CreateSockShop creates sock-shop application
-func CreateSockShop(path string, ns string) error {
-	command := exec.Command("kubectl", "apply", "-f", path, "-n", ns)
+// DeleteNamespace delete a sock-shop/loadtest namespace
+func DeleteNamespace(clientset *kubernetes.Clientset, namespaceName string) error {
```

Comment:
This has the risk of getting "stuck" if there are residual resources. This needs to be invoked once all other resources are removed in provided ns